### PR TITLE
Add edd_honeypot to the registration form validation.

### LINKED ISSUE #6115: Registration Form Honeypot Check
Hi,

The registration form (_[edd_register]_) includes a [Honeypot field](https://github.com/easydigitaldownloads/easy-digital-downloads/blob/eb881ef270d43d815754690270074940e9158faa/templates/shortcode-register.php#L43) however I see no validation of this field during the registration process.

I'm being hit with a large number of registrations despite having processes in place to protect against BOT registrations on the main WP registration forms. I guess therefore that these registrations are coming from the EDD form.

PR submitted adding a new `edd_do_honeypot_check()` function and being referenced during registration process

Thanks
Mike

### PR DESCRIPTION
**Pull request by cklosowski**
Fixes #6115 

Proposed Changes:
1. adds a check for the `edd_honeypot` field to the registration form validation.

### CODE REVIEW

**Comment by cklosowski** [comment]
To test this, you add the `[edd_register]` shortcode to a page.

As a logged out user, visit the page with the registration form.

Test 1:
Fill out the form as a normal user and it should successfully register.

Test 2: 
Before you submit the form, you need to use the Element Inspector of your browser, and add a value to the `edd_honeypot` form input. This should cause a registration form failure.

**Review by ashleyfae — APPROVED** [review]
Tested:

- [x] Normal registration without any changes. Registration was successful.
- [x] Adding value to the honeypot. Registration failed.

Does having `honeypot` in the input name reduce its effectiveness?

File: includes/login-register.php [hunk 1]
```diff
@@ -268,6 +268,10 @@ function edd_process_register_form( $data ) {
 		edd_set_error( 'payment_email_invalid', __( 'Invalid payment email', 'easy-digital-downloads' ) );
 	}
 
+	if ( isset( $data['edd_honeypot'] ) && ! empty( $data['edd_honeypot'] ) ) {
+		edd_set_error( 'invalid_form_data', __( 'Registration form validation failed', 'easy-digital-downloads' ) );
```
Comment (robincornett) [inline includes/login-register.php:272]:
```suggestion
		edd_set_error( 'invalid_form_data', __( 'Registration form validation failed.', 'easy-digital-downloads' ) );
```
I feel like with how these are displayed that they should end in a full stop. $.02

**Comment by cklosowski** [comment]
@ashleyfae 
> Does having `honeypot` in the input name reduce its effectiveness?

Honestly I don't know off hand. In my personal experience, it doesn't. This is a low hanging fruit situation and doesn't really stop someone determined to bypass basic checks. That is when we suggest things like Captchas.

**Review by robincornett — APPROVED** [review]

**Pull request merged**